OpenRA.Game/Widgets/Widget.cs

```diff
@@ -419,11 +421,25 @@ public virtual void PrepareRenderablesOuter()
 
 		public virtual void Draw() { }
 
+		void DrawDebugRenderBounds()
+		{
+			if (!Game.Settings.Debug.WidgetRenderBoundsLines)
+				return;
+
+			Game.Renderer.RgbaColorRenderer.DrawRect(
+				new float2(RenderBounds.Left, RenderBounds.Top),
+				new float2(RenderBounds.Right, RenderBounds.Bottom),
+				DebugBoundsWidth,
+				DebugBoundsColor);
+		}
+
 		public virtual void DrawOuter()
 		{
 			if (IsVisible())
 			{
 				Draw();
+				DrawDebugRenderBounds();
```

Comment:
You can save an unnecessary message call overhead in the general case by moving the settings check here.